src/WebJobs.Extensions.DurableTask/EntityQueryResult.cs

```diff
@@ -15,9 +16,25 @@ public class EntityQueryResult
     {
         internal EntityQueryResult() { }
 
-        internal EntityQueryResult(OrchestrationStatusQueryResult orchestrationResult)
+        internal EntityQueryResult(OrchestrationStatusQueryResult orchestrationResult, bool includeDeleted)
         {
-            this.Entities = orchestrationResult.DurableOrchestrationState.Select(status => new DurableEntityStatus(status));
+            if (includeDeleted)
+            {
+                this.Entities = orchestrationResult.DurableOrchestrationState
+                    .Select(status => new DurableEntityStatus(status));
```

Comment:
Do you need to add a `.ToList()` filter on the result here, like you do for the `else` case below?

Reply:
Yes, I think I should add that.